src/output_writers.jl

```diff
@@ -167,51 +161,80 @@ end
 
 function read_output(fw::NetCDFOutputWriter, field_name, iter)
     filepath = joinpath(fw.dir, filename(fw, "", iter))
-    println("[NetCDFOutputWriter] Reading fields from disk: $filepath")
     field_data = ncread(filepath, field_name)
     ncclose(filepath)
     return field_data
 end
 
-"""
-    JLD2OutputWriter(model, outputs; dir=".", prefix="", interval=1, init=noinit,
-                     force=false, asynchronous=false)
-
-Construct an `OutputWriter` that writes `label, fcn` pairs in the
-dictionary `outputs` to a single `JLD2` file, where `label` is a symbol that labels
-the output and `fcn` is a function of the form `fcn(model)` that returns
-the data to be saved. The keyword `init` is a function of the form `init(file, model)`
-that runs when the JLD2 output file is initialized.
-"""
-mutable struct JLD2OutputWriter{F, I, O} <: OutputWriter
+####
+####  JLD2 output writer
+####
+
+mutable struct JLD2OutputWriter{F, I, O, IF, IN} <: OutputWriter
         filepath :: String
          outputs :: O
         interval :: I
        frequency :: F
+            init :: IF
+       including :: IN
         previous :: Float64
-    asynchronous :: Bool
+            part :: Int
+    max_filesize :: Float64
+           async :: Bool
+           force :: Bool
+         verbose :: Bool
 end
 
 noinit(args...) = nothing
 
+"""
+    JLD2OutputWriter(model, outputs; interval=nothing, frequency=nothing, dir=".",
+                     prefix="", init=noinit, including=[:grid, :eos, :constants, :closure],
+                     part=1, max_filesize=Inf, force=false, async=false, verbose=false)
+
+Construct an `OutputWriter` that writes `label, func` pairs in the dictionary `outputs` to
+a single JLD2 file, where `label` is a symbol that labels the output and `func` is a
+function of the form `func(model)` that returns the data to be saved.
+
+# Keyword arguments
+- `frequency::Int`:    Save output every `n` model iterations.
+- `interval::Int`:     Save output every `t` units of model clock time.
+- `dir::String`:       Directory to save output to. Default: "." (current working
+                       directory).
+- `prefix::String`:    Descriptive filename prefixed to all output files. Default: "".
+- `init::Function`:    A function of the form `init(file, model)` that runs when a JLD2
+                       output file is initialized. Default: `noinit(args...) = nothing`.
+- `including::Array`:  List of model properties to save with every file. By default, the
```

Comment:
Hmmm, true it could be a `Tuple` like you say. What should I put in then? I don't think there is an abstract type for iterable collections.

I don't want to use a long `Union` so an alternative is to keep it blank and just specify it should be an iterable collecting.